ovs/lib/albacontroller.py

```diff
@@ -617,201 +710,220 @@ def nsm_checkup(allow_offline=False, backend_guid=None):
-                                                  service_type=nsm_service_type,
-                                                  ports=[nsm_result['client_port'], nsm_result['messaging_port']],
-                                                  storagerouter=storagerouter,
-                                                  junction_type=NSMService,
-                                                  backend=backend,
-                                                  number=maxnumber)
-            for storagerouter in storagerouters:
-                client = SSHClient(storagerouter, username='root')
-                ArakoonInstaller.start(nsm_name, client)
-            AlbaController.register_nsm(abm_name=abm_service_name,
-                                        nsm_name=nsm_name,
-                                        ip=storagerouters[0].ip)
-            logger.debug('New NSM ({0}) added'.format(maxnumber))
-        else:
-            logger.debug('NSM load OK')
+                    maxnumber = max(nsm_loads.keys()) + 1
+                    if nsm_metadata.internal is False:
+                        logger.debug('Externally managed NSM arakoon cluster needs to be expanded')
+                        clusters = ArakoonInstaller.get_arakoon_metadata_by_cluster_type(cluster_type=ServiceType.ARAKOON_CLUSTER_TYPES.NSM, in_use=False)
+                        if len(clusters) == 0:
+                            logger.warning('Cannot claim additional NSM clusters, because no clusters are available')
+                        else:
+                            client = None
+                            masters = StorageRouterList.get_masters()
+                            for master in masters:
+                                try:
+                                    client = SSHClient(master)
+                                    break
+                                except UnableToConnectException:
+                                    continue
+                            if client is None:
+                                raise ValueError('Could not find an online master node')
+                            AlbaController._model_service(service_name=clusters[0].cluster_id,
+                                                          service_type=nsm_service_type,
+                                                          ports=[],
+                                                          storagerouter=None,
+                                                          junction_type=NSMService,
+                                                          backend=alba_backend,
+                                                          number=maxnumber)
+                            AlbaController.register_nsm(abm_name=abm_service_name,
+                                                        nsm_name=clusters[0].cluster_id,
+                                                        ip=client.ip)
+                            clusters[0].claim()
```

Comment:
Make sure that you validate whether the returned metadata is of an external cluster.